### PR TITLE
87 add csrf tokens to forms

### LINKED ISSUE #87: Add CSRF tokens to forms
CSRF tokens are used to protect against malicious form posts. This is done by sending a "token" on form submit, which is earlier supplied by the server. These tokens have to correspond.

## Technical steps
* Research if we want to use CSRF for all requests or forms only, and also if the latter is supported when using HTMX
* CSRF tokens are sent and received as headers, and checked (only) on form submission
* https://pypi.org/project/fastapi-csrf-protect/

### PR DESCRIPTION
**Pull request by ChristopherSpelt**
# Add CSRF Tokens to projects/new post request

Adds CSRF protection via [fastapi CSRF Token Protect](https://pypi.org/project/fastapi-csrf-protect/) (implementing double submit cookies) to the `projects/new` post request as a middleware layer.

Resolves #87 

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x]  I have tested the changes locally and verified that they work as expected.
- [x]  I have followed the project's coding conventions and style guidelines.
- [x]  I have rebased my branch onto the latest commit of the main branch.
- [x]  I have squashed or reorganized my commits into logical units.
- [x]  I have read, understood and agree to the [Developer Certificate of Origin](../blob/main/DCO.md), which this project utilizes.

### CODE REVIEW

File: amt/api/routes/projects.py [hunk 1]
```diff
@@ -35,18 +37,31 @@ async def get_root(
...
 ) -> HTMLResponse:
+    csrf_token, signed_token = csrf_protect.generate_csrf_tokens()
+
     instruments = instrument_service.fetch_instruments()
+    response = templates.TemplateResponse(
+        request, "projects/new.html.j2", {"csrf_token": csrf_token, "instruments": instruments}
+    )
 
-    return templates.TemplateResponse(request, "projects/new.html.j2", {"instruments": instruments})
+    csrf_protect.set_csrf_cookie(signed_token, response)
```
[the remark this answers is not in the document]
Comment (berrydenhartog) [reply amt/api/routes/projects.py:49]: you could also consider using the samesite cookie with a session if you prefer that.

File: amt/core/csrf.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+from amt.core.config import get_settings
+
+
+@CsrfProtect.load_config  # type: ignore
+def get_csrf_config() -> list[tuple[Any, ...]]:
+    settings = get_settings()
+    config = [
+        ("secret_key", settings.CSRF_PROTECT_SECRET_KEY),
+        ("token_location", settings.CSRF_TOKEN_LOCATION),
+        ("token_key", settings.CSRF_TOKEN_KEY),
```
[the remark this answers is not in the document]
Comment (ChristopherSpelt) [reply amt/core/csrf.py:14]: Set cookie_samesite to strict. Default cookie age is 3600 seconds

File: amt/api/routes/projects.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChristopherSpelt) [reply amt/api/routes/projects.py]: Added as middleware function

**Review by berrydenhartog — APPROVED** [review]
LGTM

**Pull request merged**